### ISSUE TITLE
Caldav Issue (SoGo?)

### ISSUE THREAD

**Issue by fairlygood** [mailcow/mailcow-dockerized#1689]
Morning,

I noticed last week that my calendars and reminders no longer sync over caldav. They work fine over ActiveSync but I can't use this on OSX.

The calendar app reports:

```
'The request (CalDAVAccountRefreshQueueableOperation) for account “jamieadams.me.uk” failed.'
```

Not sure where to start debugging but I've updated the server, restarted containers, removed and added the device accounts. The symptoms are the same using Caldav on my Mac / iPhone & iPad. Contacts seems to sync fine - just not calendar and reminders.

I can't see anything useful in the SoGo log, but here it is:

```

-- | -- | --
18/08/2018, 20:15:33 | notice | [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/Contacts/jamieadams.me.uk/ HTTP/1.0" 207 511/717 0.007 1380 62% 0
18/08/2018, 20:15:33 | notice | [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/Contacts/ HTTP/1.0" 207 658/717 0.011 2862 77% 0
18/08/2018, 20:15:33 | notice | [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 207 566/439 0.005 2159 73% 0
18/08/2018, 20:15:33 | notice | [53]: 46.231.13.136 "OPTIONS /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 200 0/0 0.004 - - 0
18/08/2018, 20:15:33 | notice | [53]: 46.231.13.136 "OPTIONS /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 401 0/0 0.005 - - 0
18/08/2018, 20:15:10 | notice | [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/Calendar/ HTTP/1.0" 207 2475/2217 0.045 40972 93% 0
18/08/2018, 20:15:10 | notice | [53]: 46.231.13.136 "REPORT /SOGo/dav/ HTTP/1.0" 207 300/89 0.003 1210 75% 0
18/08/2018, 20:15:10 | notice | [53]: 46.231.13.136 "OPTIONS /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 200 0/0 0.003 - - 0
18/08/2018, 20:15:10 | notice | [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 207 655/743 0.005 2873 77% 0
18/08/2018, 20:15:10 | notice | [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/Calendar/ HTTP/1.0" 207 2475/2217 0.043 40972 93% 0
18/08/2018, 20:15:10 | notice | [53]: 46.231.13.136 "REPORT /SOGo/dav/ HTTP/1.0" 207 300/89 0.003 1210 75% 0
18/08/2018, 20:15:10 | notice | [53]: 46.231.13.136 "REPORT /SOGo/dav/ HTTP/1.0" 401 0/89 0.002 - - 0
18/08/2018, 20:15:10 | notice | [53]: 46.231.13.136 "OPTIONS /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 200 0/0 0.002 - - 0
18/08/2018, 20:15:10 | notice | [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 207 655/743 0.006 2873 77% 0
18/08/2018, 20:15:10 | notice | [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 401 0/743 0.002 - - 0
18/08/2018, 20:12:43 | notice | [53]: 46.231.13.136 "REPORT /SOGo/dav/hello%40jamieadams.me.uk/Contacts/personal/ HTTP/1.0" 207 125/210 0.008 - - 0
18/08/2018, 20:12:43 | notice | [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/Contacts/jamieadams.me.uk/ HTTP/1.0" 207 511/717 0.006 1380 62% 0
18/08/2018, 20:12:43 | notice | [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/Contacts/ HTTP/1.0" 207 658/717 0.010 2862 77% 0
```

Not sure if the logs from the osx console are any use but here they are:

```
default	20:15:10.056795 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.queue] [<CalDAVAccountRefreshQueueableOperation: 0x7ff0a746a9d0; Sequence: 0> -  Refreshing properties for principal jamieadams.me.uk]
default	20:15:10.058935 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.http] [PROPFIND https://hello%40jamieadams.me.uk@mercury.iofoundry.eu/SOGo/dav/hello%40jamieadams.me.uk/]
default	20:15:10.069427 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.http] [REPORT https://hello%40jamieadams.me.uk@p36-caldav.icloud.com/195762511/calendars/]
default	20:15:10.380092 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.http] [OPTIONS https://hello%40jamieadams.me.uk@mercury.iofoundry.eu/SOGo/dav/hello%40jamieadams.me.uk/]
default	20:15:10.442670 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.http] [REPORT https://hello%40jamieadams.me.uk@mercury.iofoundry.eu/SOGo/dav/]
default	20:15:10.533941 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.queue] [<CalDAVAccountRefreshQueueableOperation: 0x7ff0a746a9d0; Sequence: 0> -  Refreshing calendars for principal jamieadams.me.uk]
default	20:15:10.534394 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.queue] [Principal jamieadams.me.uk doesn't have sync token and/or doesn't support calendar home sync (0).  Doing regular refresh. Deep refresh? 1]
default	20:15:10.535604 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.http] [PROPFIND https://hello%40jamieadams.me.uk@mercury.iofoundry.eu/SOGo/dav/hello%40jamieadams.me.uk/Calendar/]
default	20:15:10.622796 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.queue] [Account refresh failed with error: Error Domain=CoreDAVErrorDomain Code=3 "(null)" UserInfo={AccountName=jamieadams.me.uk, CalDAVErrFromRefresh=YES}]
default	20:15:10.628586 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.queue] [<CalDAVAccountRefreshQueueableOperation: 0x7ff0a7789ad0; Sequence: 0> -  Refreshing properties for principal jamieadams.me.uk]
default	20:15:10.629041 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.http] [PROPFIND https://hello%40jamieadams.me.uk@mercury.iofoundry.eu/SOGo/dav/hello%40jamieadams.me.uk/]
default	20:15:10.675734 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.http] [OPTIONS https://hello%40jamieadams.me.uk@mercury.iofoundry.eu/SOGo/dav/hello%40jamieadams.me.uk/]
default	20:15:10.723474 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.http] [REPORT https://hello%40jamieadams.me.uk@mercury.iofoundry.eu/SOGo/dav/]
default	20:15:10.838158 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.queue] [<CalDAVAccountRefreshQueueableOperation: 0x7ff0a7789ad0; Sequence: 0> -  Refreshing calendars for principal jamieadams.me.uk]
default	20:15:10.838465 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.queue] [Principal jamieadams.me.uk doesn't have sync token and/or doesn't support calendar home sync (0).  Doing regular refresh. Deep refresh? 1]
default	20:15:10.839694 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.http] [PROPFIND https://hello%40jamieadams.me.uk@mercury.iofoundry.eu/SOGo/dav/hello%40jamieadams.me.uk/Calendar/]
default	20:15:10.939908 +0100	CalendarAgent	[com.apple.calendar.store.log.caldav.queue] [Account refresh failed with error: Error Domain=CoreDAVErrorDomain Code=3 "(null)" UserInfo={AccountName=jamieadams.me.uk, CalDAVErrFromRefresh=YES}]
```

**Comment by fairlygood** [comment]
I solved this by backing up my data, erasing the mallow installation, reinstalling and restoring data.

I've no idea what caused it, but it's fixed now.

**Comment by fairlygood** [comment]
Sadly this seems to have reoccured and calendars / reminders will not sync over Caldav anymore.

**Comment by andryyy** [comment]
Can you check sogo-mailcow logs? Any logs lines like "no childs available"?

**Comment by fairlygood** [comment]
Hi, no nothing like that.

The lines in the log relating to my account look similar to those I posted above from the web interface:

```
Aug 18 20:15:10 bd11af63c278 sogod [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 401 0/743 0.002 - - 0
Aug 18 20:15:10 bd11af63c278 sogod [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 207 655/743 0.006 2873 77% 0
Aug 18 20:15:10 bd11af63c278 sogod [53]: 46.231.13.136 "OPTIONS /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 200 0/0 0.002 - - 0
Aug 18 20:15:10 bd11af63c278 sogod [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/Calendar/ HTTP/1.0" 207 2475/2217 0.043 40972 93% 0
Aug 18 20:15:10 bd11af63c278 sogod [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 207 655/743 0.005 2873 77% 0
Aug 18 20:15:10 bd11af63c278 sogod [53]: 46.231.13.136 "OPTIONS /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 200 0/0 0.003 - - 0
Aug 18 20:15:10 bd11af63c278 sogod [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/Calendar/ HTTP/1.0" 207 2475/2217 0.045 40972 93% 0
Aug 18 20:15:30 bd11af63c278 sogod [56]: <0x0x557a78835d40[SOGoActiveSyncDispatcher]> Sleeping 30 seconds while detecting changes for user hello@jamieadams.me.uk in Ping...
Aug 18 20:15:33 bd11af63c278 sogod [53]: 46.231.13.136 "OPTIONS /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 401 0/0 0.005 - - 0
Aug 18 20:15:33 bd11af63c278 sogod [53]: 46.231.13.136 "OPTIONS /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 200 0/0 0.004 - - 0
Aug 18 20:15:33 bd11af63c278 sogod [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/ HTTP/1.0" 207 566/439 0.005 2159 73% 0
Aug 18 20:15:33 bd11af63c278 sogod [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/Contacts/ HTTP/1.0" 207 658/717 0.011 2862 77% 0
Aug 18 20:15:33 bd11af63c278 sogod [53]: 46.231.13.136 "PROPFIND /SOGo/dav/hello%40jamieadams.me.uk/Contacts/jamieadams.me.uk/ HTTP/1.0" 207 511/717 0.007 1380 62% 0
Aug 18 20:15:33 bd11af63c278 sogod [53]: 46.231.13.136 "REPORT /SOGo/dav/hello%40jamieadams.me.uk/Contacts/personal/ HTTP/1.0" 207 615/236 0.014 13296 95% 0
```

Lot's of these though (I don't know if these are normal): 
```
Aug 18 20:16:26 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b2e580[WOWatchDogChild]> pid 56 has been hanging in the same request for 5 minutes
Aug 18 20:16:57 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b1b300[WOWatchDogChild]> pid 51 has been hanging in the same request for 3 minutes
Aug 18 20:17:57 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b1b300[WOWatchDogChild]> pid 51 has been hanging in the same request for 4 minutes
Aug 18 20:18:03 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b2e580[WOWatchDogChild]> pid 56 has been hanging in the same request for 1 minutes
Aug 18 20:18:57 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b1b300[WOWatchDogChild]> pid 51 has been hanging in the same request for 5 minutes
Aug 18 20:19:03 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b2e580[WOWatchDogChild]> pid 56 has been hanging in the same request for 2 minutes
Aug 18 20:20:02 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b12e70[WOWatchDogChild]> pid 53 has been hanging in the same request for 1 minutes
Aug 18 20:20:03 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b2e580[WOWatchDogChild]> pid 56 has been hanging in the same request for 3 minutes
Aug 18 20:21:02 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b12e70[WOWatchDogChild]> pid 53 has been hanging in the same request for 2 minutes
Aug 18 20:21:03 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b2e580[WOWatchDogChild]> pid 56 has been hanging in the same request for 4 minutes
Aug 18 20:22:02 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b12e70[WOWatchDogChild]> pid 53 has been hanging in the same request for 3 minutes
Aug 18 20:22:03 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b2e580[WOWatchDogChild]> pid 56 has been hanging in the same request for 5 minutes
Aug 18 20:23:02 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b12e70[WOWatchDogChild]> pid 53 has been hanging in the same request for 4 minutes
Aug 18 20:23:40 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b2e580[WOWatchDogChild]> pid 56 has been hanging in the same request for 1 minutes
Aug 18 20:24:02 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b12e70[WOWatchDogChild]> pid 53 has been hanging in the same request for 5 minutes
Aug 18 20:24:40 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b2e580[WOWatchDogChild]> pid 56 has been hanging in the same request for 2 minutes
Aug 18 20:24:58 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b1b300[WOWatchDogChild]> pid 51 has been hanging in the same request for 1 minutes
Aug 18 20:25:40 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b2e580[WOWatchDogChild]> pid 56 has been hanging in the same request for 3 minutes
Aug 18 20:25:58 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b1b300[WOWatchDogChild]> pid 51 has been hanging in the same request for 2 minutes
Aug 18 20:26:40 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b2e580[WOWatchDogChild]> pid 56 has been hanging in the same request for 4 minutes
Aug 18 20:26:58 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b1b300[WOWatchDogChild]> pid 51 has been hanging in the same request for 3 minutes
Aug 18 20:27:40 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b2e580[WOWatchDogChild]> pid 56 has been hanging in the same request for 5 minutes
Aug 18 20:27:58 bd11af63c278 sogod [10]: [WARN] <0x0x557a77b1b300[WOWatchDogChild]> pid 51 has been hanging in the same request for 4 minutes
```

**Comment by fairlygood** [comment]
I rebooted the machine, because I'm a full believer in the 'switch it off and on again' method. Here's my full sogo log from boot: https://pastebin.com/J7kEdaPp

**Comment by andryyy** [comment]
Have you tried to create a second mailbox? Does it sync? Could be a "broken" DAV record.

**Comment by fairlygood** [comment]
Hi andryyy,

Thanks for taking the time to respond and help me debug.

I've started over from scratch with the server - which was surprising quick and easy - and it seems to be working for now. I had some emojis in my calendar names which I guess the server didn't seem to like (they kept changing / reverting between devices). I only added those recently so might be linked. It's weird, because it's been ticking over for the last year or so with no issues!

I didn't actually check if it was just my devices affected. I'll check that if it does reoccur.

I'll close for now and keep my fingers crossed.

Thanks again.
